components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/cache/AuthorizationGrantCache.java

```diff
@@ -87,9 +99,23 @@ public void addToCacheByToken(AuthorizationGrantCacheKey key, AuthorizationGrant
      * @return Cached entry.
      */
     public AuthorizationGrantCacheEntry getValueFromCacheByToken(AuthorizationGrantCacheKey key) {
+
+        if (log.isDebugEnabled() && key != null) {
+            log.debug("Getting from cache by token - " + key.getUserAttributesId());
+        }
         AuthorizationGrantCacheEntry cacheEntry = super.getValueFromCache(key);
+        if (log.isDebugEnabled() && cacheEntry != null) {
+            log.debug("Got value from cache.");
```

Comment:
This log would be meaningless in a multi-threaded scenario. Can we add some context info?